tests/base/test_array.py

```diff
@@ -0,0 +1,467 @@
+"""Tests for array interface base functions."""
+
+# pylint: disable=no-member, redefined-outer-name, no-self-use, protected-access
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose, assert_array_equal
+
+from arviz_stats.base.array import BaseArray, process_ary_axes, process_chain_none
+
+
+@pytest.fixture
+def array_stats():
+    return BaseArray()
+
+
+class TestHelperFunctions:
+    def test_process_chain_none_with_none(self):
+        ary = np.random.randn(100, 50)
```

Comment:
these should switch to using the newer numpy random generators. Using `RandomState` instead of `Generator` is discouraged, calling `numpy.random.function` directly even more so. Ref: https://numpy.org/doc/stable/reference/random/legacy.html#functions-in-numpy-random

EDIT: in this particular function and a few more below where we basically care about the shape only we are probably better off using `np.empty` though.